/home/janneke/vc/gub/target/linux-64/root/usr/include/pngconf.h:307: error: expected constructor, destructor, or type conversion before '.' token
/home/janneke/vc/gub/target/linux-64/root/usr/include/pngconf.h:308: error: '__dont__' does not name a type
make[2]: *** [sp-image.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory `/home/janneke/vc/gub/target/linux-64/build/inkscape-trunk-20605/src'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/home/janneke/vc/gub/target/linux-64/build/inkscape-trunk-20605'
make: *** [all] Error 2

Probably an inkscape bug,
See also: https://bugs.launchpad.net/ubuntu/+source/libpng/+bug/218409

$ diff -purN libpng-1.2.8.config/pngconf.h.orig libpng-1.2.8.config/pngconf.h
--- libpng-1.2.8.config/pngconf.h.orig	2004-12-03 01:14:24.000000000 +0100
+++ libpng-1.2.8.config/pngconf.h	2009-02-02 15:42:16.000000000 +0100
@@ -304,8 +304,8 @@
      /* If you encounter a compiler error here, see the explanation
       * near the end of INSTALL.
       */
-         __png.h__ already includes setjmp.h;
-         __dont__ include it again.;
+#warning __png.h__ already includes setjmp.h;
+#warning __dont__ include it again.;
 #    endif
 #  endif /* __linux__ */
 
